src/med_bench/utils/classifiers.py

```diff
@@ -0,0 +1,423 @@
+        f_m1x[test_index] = fm_1[test_ind, m[test_index]]
+
+        # predict E[Y|T=t,M,X]
+        mu_1mx[test_index] = clf_y.predict(x_t1_m[test_index, :])
+        mu_0mx[test_index] = clf_y.predict(x_t0_m[test_index, :])
+
+        # predict E[Y|T=t,M=m,X]
+        mu_00x[test_index] = clf_y.predict(x_t0_m0[test_index, :])
+        mu_01x[test_index] = clf_y.predict(x_t0_m1[test_index, :])
+        mu_11x[test_index] = clf_y.predict(x_t1_m1[test_index, :])
+        mu_10x[test_index] = clf_y.predict(x_t1_m0[test_index, :])
+
+        # E[E[Y|T=1,M=m,X]|T=t,X] model fitting
+        reg_y_t1m1_t0 = clone(classifier_cross_y).fit(
+            x[test_index, :][ind_t0, :], mu_11x[test_index][ind_t0]
+        )
+        reg_y_t1m0_t0 = clone(classifier_cross_y).fit(
+            x[test_index, :][ind_t0, :], mu_10x[test_index][ind_t0]
+        )
+        reg_y_t1m1_t1 = clone(classifier_cross_y).fit(
+            x[test_index, :][~ind_t0, :], mu_11x[test_index][~ind_t0]
+        )
+        reg_y_t1m0_t1 = clone(classifier_cross_y).fit(
+            x[test_index, :][~ind_t0, :], mu_10x[test_index][~ind_t0]
+        )
+
+        # predict E[E[Y|T=1,M=m,X]|T=t,X]
+        E_mu_t1_t0[test_index] = (
+                reg_y_t1m0_t0.predict(x[test_index, :]) * f_00x[test_index]
+                + reg_y_t1m1_t0.predict(x[test_index, :]) * f_01x[test_index]
+        )
+        E_mu_t1_t1[test_index] = (
+                reg_y_t1m0_t1.predict(x[test_index, :]) * f_10x[test_index]
+                + reg_y_t1m1_t1.predict(x[test_index, :]) * f_11x[test_index]
+        )
+
+        # E[E[Y|T=0,M=m,X]|T=t,X] model fitting
+        reg_y_t0m1_t0 = clone(classifier_cross_y).fit(
+            x[test_index, :][ind_t0, :], mu_01x[test_index][ind_t0]
+        )
+        reg_y_t0m0_t0 = clone(classifier_cross_y).fit(
+            x[test_index, :][ind_t0, :], mu_00x[test_index][ind_t0]
+        )
+        reg_y_t0m1_t1 = clone(classifier_cross_y).fit(
+            x[test_index, :][~ind_t0, :], mu_01x[test_index][~ind_t0]
+        )
+        reg_y_t0m0_t1 = clone(classifier_cross_y).fit(
+            x[test_index, :][~ind_t0, :], mu_00x[test_index][~ind_t0]
+        )
+
+        # predict E[E[Y|T=0,M=m,X]|T=t,X]
+        E_mu_t0_t0[test_index] = (
+                reg_y_t0m0_t0.predict(x[test_index, :]) * f_00x[test_index]
+                + reg_y_t0m1_t0.predict(x[test_index, :]) * f_01x[test_index]
+        )
+        E_mu_t0_t1[test_index] = (
+                reg_y_t0m0_t1.predict(x[test_index, :]) * f_10x[test_index]
+                + reg_y_t0m1_t1.predict(x[test_index, :]) * f_11x[test_index]
+        )
+
+    # f = f_00x, f_01x, f_10x, f_11x, f_m0x, f_m1x
+    f = f_m0x, f_m1x
+    # mu = mu_11x, mu_10x, mu_01x, mu_00x, mu_t0, mu_t1
+    mu = mu_0mx, mu_1mx
+    cross_mu = E_mu_t0_t0, E_mu_t0_t1, E_mu_t1_t0, E_mu_t1_t1
+
+    return p_x, f, mu, cross_mu
+
+def get_x_y_classifiers(regularization, forest, calib_method, random_state):
```

Comment:
Same here regarding docstring

Reply:
This has been addressed in 5c5d7c0b812d2aa64f6ae01e519138b341c2f8a5 .